crates/nu-parser/src/parse_keywords.rs

```diff
@@ -134,97 +134,118 @@ pub fn parse_keyword(
 }
 
 pub fn parse_def_predecl(working_set: &mut StateWorkingSet, spans: &[Span]) {
-    let name = working_set.get_span_contents(spans[0]);
+    let mut pos = 0;
+
+    let def_type_name = if spans.len() >= 3 {
+        // definition can't have only two spans, minimum is 3, e.g., 'extern spam []'
+        let first_word = working_set.get_span_contents(spans[0]);
+
+        if first_word == b"export" {
+            pos += 2;
+        } else {
+            pos += 1;
+        }
 
-    // handle "export def" same as "def"
-    let (decl_name, spans) = if name == b"export" && spans.len() >= 2 {
-        (working_set.get_span_contents(spans[1]), &spans[1..])
+        working_set.get_span_contents(spans[pos - 1]).to_vec()
     } else {
-        (name, spans)
+        return;
```

Comment:
Is the idea here to return early and catch the error later?

How do we report an error for an incomplete `def`?

Reply:
Yeah, you'll get a missing positional error during the actual parse. During the predecl scan you only want to predeclare valid commands but don't want to repeat the logic of the parser. I just tested it, it beehaves similarly to the old version.